Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,244 @@
 #include "simbody/internal/GeneralForceSubsystem.h"
 #include "simbody/internal/SimbodyMatterSubsystem.h"
 #include "simbody/internal/MultibodySystem.h"
+#include <iostream>
+#include <exception>
+#include <mutex>
+#include <thread>
 
 #include "ForceImpl.h"
 
+#include <memory>
+
+#define NUMNONPARALLELTHREADS 1
+#define NONPARALLELFORCESINDEX 0
 
 namespace SimTK {
 
+/** Calls calcForce() on all enabled forces. For an index of 0, this Task
+ calculates the force for all non-parallelized forces. For all other indices,
+ this Task executes a single parallelized force. */
```

Comment:
There is clearly some tricky stuff going on here with several different modes including "cached and non cached" but it wasn't clear what all that means. Please add some additional explanation either in the class description or with the individual methods and enums.